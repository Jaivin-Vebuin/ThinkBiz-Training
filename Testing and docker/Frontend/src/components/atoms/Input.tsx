import React from "react";

interface InputProps {
  name: string;
  value?: string;
  handleOnChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  type?: string;
  placeholder?: string;
  inputType: string;
  labelText: string;
  htmlFor: string;
  registerProps?: any;
  error?: string;
}

const Input: React.FC<InputProps> = ({
  name,
  value,
  handleOnChange,
  inputType,
  placeholder,
  labelText,
  htmlFor,
  registerProps,
  error,
}) => {
  return (
    <div>
      <label htmlFor={htmlFor}>{labelText}</label>
      <input
        id={htmlFor}
        name={name}
        value={value}
        onChange={handleOnChange}
        type={inputType}
        placeholder={placeholder}
        {...registerProps}
      />
      {error && <p style={{ color: "red" }}>{error}</p>}{" "}
    </div>
  );
};

export default Input;
