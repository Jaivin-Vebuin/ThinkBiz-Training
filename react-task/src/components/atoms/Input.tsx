import React from "react";
import { ErrorFieldStyle } from "../styles/AuthFormStyles";

interface InputProps {
  name: string;
  value?: string;
  handleOnChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  type?: string;
  placeholder?: string;
  inputType: string;
  labelText: string;
  htmlFor: string;
  registerProps?: any;
  error?: string;
}

const Input: React.FC<InputProps> = ({
  name,
  value,
  handleOnChange,
  inputType,
  placeholder,
  labelText,
  htmlFor,
  registerProps,
  error,
}) => {
  return (
    <div>
      <label htmlFor={htmlFor}>{labelText}</label>
      <input
        name={name}
        value={value}
        onChange={handleOnChange}
        type={inputType}
        placeholder={placeholder}
        {...registerProps}
      />
      {error && (
        <ErrorFieldStyle>
          <p>{error}</p>
        </ErrorFieldStyle>
      )}
    </div>
  );
};

export default Input;
