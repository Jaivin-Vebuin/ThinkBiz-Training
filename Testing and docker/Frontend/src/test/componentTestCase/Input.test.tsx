import "@testing-library/jest-dom";
import { render, fireEvent } from "@testing-library/react";
import Input from "../../components/atoms/Input";

describe("Input Component", () => {
  test("renders input with label and handles change event", () => {
    const handleChange = jest.fn();
    const { getByLabelText, asFragment } = render(
      <Input 
        name="test-input" 
        inputType="text" 
        labelText="Test Label" 
        htmlFor="test-input" 
        handleOnChange={handleChange} 
      />
    );
    
    const inputElement = getByLabelText("Test Label", { selector: "input" });
    expect(inputElement).toBeInTheDocument();
    
    fireEvent.change(inputElement, { target: { value: "new value" } });
    expect(handleChange).toHaveBeenCalledTimes(1);
    
    expect(asFragment()).toMatchSnapshot();
  });
});
