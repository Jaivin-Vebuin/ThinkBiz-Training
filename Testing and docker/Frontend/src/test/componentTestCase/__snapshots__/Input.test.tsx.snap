// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders input with label and handles change event 1`] = `
<DocumentFragment>
  <div>
    <label
      for="test-input"
    >
      Test Label
    </label>
    <input
      id="test-input"
      name="test-input"
      type="text"
    />
     
  </div>
</DocumentFragment>
`;
